src/legacy/ui/public/agg_types/buckets/_bucket_agg_type.ts

```diff
@@ -17,13 +17,32 @@
  * under the License.
  */
 
-import { AggType, AggTypeConfig } from '../agg_type';
+import { AggParamType } from '../param_types/agg';
 import { AggConfig } from '../../vis';
+import { KBN_FIELD_TYPES } from '../../../../../plugins/data/common';
+import { AggType, AggTypeConfig } from '../agg_type';
+
+export type IBucketAggConfig = AggConfig;
+
+export interface BucketAggParam extends AggParamType {
+  scriptable?: boolean;
+  filterFieldTypes?: KBN_FIELD_TYPES | KBN_FIELD_TYPES[] | '*';
+}
 
-export class BucketAggType extends AggType {
-  getKey: (bucket: any, key: any, agg: AggConfig) => any;
+export interface BucketAggTypeConfig<TBucketAggConfig extends IBucketAggConfig>
```

Comment:
I would leave some comments here as well, incl. why is this needed and what should it be replaced with